src/NServiceBus.AcceptanceTests/Sagas/When_simple_saga.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace NServiceBus.AcceptanceTests.Sagas
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_simple_saga : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_execute_saga()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<SagaEndpoint>(b => b
+                    .When(session => session.SendLocal(new StartSagaMessage
+                    {
+                        Key = "Part1_Part2"
+                    }))
+                    .When(c => c.FirstMessageReceived, session => session.SendLocal(new OtherMessage
+                    {
+                        Part1 = "Part1",
+                        Part2 = "Part2"
+                    })))
+                .Done(c => c.SecondMessageReceived)
+                .Run();
+
+            Assert.IsTrue(context.SecondMessageReceived);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool FirstMessageReceived { get; set; }
+            public bool SecondMessageReceived { get; set; }
+            public Guid SagaId { get; set; }
+        }
+
+        public class SagaEndpoint : EndpointConfigurationBuilder
+        {
+            public SagaEndpoint()
+            {
+                //note: the concurrency checks for the InMemory persister doesn't seem to work so limiting to 1 for now
```

Comment:
what do you mean by concurrency checks don't work? I think the issue is that the flag which triggers the second message to be dispatched can be set before the message is completely processed and therefore not fully commited when the second message arrives? Would it be possible to dispatch `OtherMessage` from the `StartSagaMessage ` handler?